pkg/storage/storage.go

```diff
@@ -0,0 +1,191 @@
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/google/uuid"
+	_ "github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/carfloresf/the-sun-god/config"
+	"github.com/carfloresf/the-sun-god/internal/model"
+)
+
+const (
+	insertPostQuery = `INSERT
+						INTO post(id,title,author,subreddit,link,content,score,promoted,nsfw) 
+						VALUES    ($1, $2, $3, $4, $5, $6, $7, $8, $9)`
+	updatePostScoreQuery = `UPDATE post SET score = $1 WHERE id = $2`
+	countTotalPostsQuery = `SELECT COUNT(*) FROM post WHERE promoted = false`
+	queryPosts           = `SELECT id,title,author,subreddit,link,content,score,promoted,nsfw,created_at 
+							FROM post WHERE promoted = $1 ORDER BY score DESC, created_at DESC LIMIT $2 OFFSET $3;`
+	queryPromotedPosts = `SELECT id,title,author,subreddit,link,content,score,promoted,nsfw,created_at 
+							FROM post WHERE promoted = true ORDER BY RANDOM() LIMIT $1;`
+)
+
+var allStatements = []string{insertPostQuery, updatePostScoreQuery, countTotalPostsQuery, queryPosts, queryPromotedPosts}
```

Comment:
If you are using `allStatements` only for convenience of preparing them, maybe you should move it to `PrepareAllStatements`

Preparing all statements should be done once, and we don't need to have this variable live the entire life of the program

Reply:
Yeah good point